tests/ops/op_math/test_linear_combination.py

```diff
@@ -0,0 +1,2032 @@
+                qml.ops.LinearCombination(
+                    coeffs,
+                    [X(0), Z(0)],
+                    simplify=simplify,
+                    grouping_type=group,
+                )
+            )
+
+        with tf.GradientTape() as tape:
+            res = circuit(coeffs, param)
+        grad = tape.gradient(res, [coeffs, param])
+
+        # differentiating a cost that combines circuits with
+        # measurements expval(Pauli)
+
+        # we need to create new tensors here
+        coeffs2 = tf.Variable([-0.05, 0.17], dtype=tf.double)
+        param2 = tf.Variable(1.7, dtype=tf.double)
+        half1 = qml.QNode(circuit1, dev, interface="tf", diff_method="backprop")
+        half2 = qml.QNode(circuit2, dev, interface="tf", diff_method="backprop")
+
+        def combine(coeffs, param):
+            return coeffs[0] * half1(param) + coeffs[1] * half2(param)
+
+        with tf.GradientTape() as tape2:
+            res_expected = combine(coeffs2, param2)
+        grad_expected = tape2.gradient(res_expected, [coeffs2, param2])
+
+        assert np.allclose(grad[0], grad_expected[0])
+        assert np.allclose(grad[1], grad_expected[1])
+
+    @pytest.mark.tf
+    def test_nontrainable_coeffs_tf(self):
+        """Test the tf interface if the coefficients are explicitly set non-trainable"""
+
+        coeffs = tf.constant([-0.05, 0.17], dtype=tf.double)
+        param = tf.Variable(1.7, dtype=tf.double)
+
+        # differentiating a circuit with measurement expval(H)
+        @qml.qnode(dev, interface="tf", diff_method="backprop")
+        def circuit(coeffs, param):
+            qml.RX(param, wires=0)
+            qml.RY(param, wires=0)
+            return qml.expval(
+                qml.ops.LinearCombination(
+                    coeffs,
+                    [X(0), Z(0)],
+                )
+            )
+
+        with tf.GradientTape() as tape:
+            res = circuit(coeffs, param)
+        grad = tape.gradient(res, [coeffs, param])
+
+        # differentiating a cost that combines circuits with
+        # measurements expval(Pauli)
+
+        # we need to create new tensors here
+        coeffs2 = tf.constant([-0.05, 0.17], dtype=tf.double)
+        param2 = tf.Variable(1.7, dtype=tf.double)
+        half1 = qml.QNode(circuit1, dev, interface="tf", diff_method="backprop")
+        half2 = qml.QNode(circuit2, dev, interface="tf", diff_method="backprop")
+
+        def combine(coeffs, param):
+            return coeffs[0] * half1(param) + coeffs[1] * half2(param)
+
+        with tf.GradientTape() as tape2:
+            res_expected = combine(coeffs2, param2)
+        grad_expected = tape2.gradient(res_expected, [coeffs2, param2])
+
+        assert grad[0] is None
+        assert np.allclose(grad[1], grad_expected[1])
+
+    # TODO: update logic of adjoint differentiation to catch attempt to differentiate lincomb coeffs
+    @pytest.mark.xfail
+    def test_not_supported_by_adjoint_differentiation(self):
```

Comment:
If the test isn't failing with adjoint and the results are correct, this test should be removed and replaced with tests for correctness with adjoint.

Reply:
Yes, that is actually done on `ham-tests`, should have put that here and pushed upstream